CRM/SMS/BAO/Provider.php

```diff
@@ -93,7 +93,9 @@ public static function getProviders($selectArr = NULL, $filter = NULL, $getActiv
   public static function saveRecord($values) {
     $dao = new CRM_SMS_DAO_Provider();
     $dao->copyValues($values);
-    $dao->domain_id = CRM_Core_Config::domainID();
+    if (!empty($dao->domain_id)) {
```

Comment:
- oh it's a domain-theft pattern? I edit something on another domain & I steal the record? That seems like maybe a form-layer thing to me